client/src/wayland_obj/wl_compositor.rs

```diff
@@ -1,37 +1,43 @@
 //! Wrapper around a wl_compositor.
 
-use std::cell::RefCell;
-
 use wayland_client::{
     protocol::{wl_compositor::WlCompositor, wl_surface::WlSurface},
     GlobalImplementor, NewProxy
 };
 
 /// The minimum version of the wl_compositor global to bind to.
-pub const WL_COMPOSITOR_VERSION: u32 = 3;
+pub(super) const WL_COMPOSITOR_VERSION: u32 = 3;
+
+pub type Surface = WlSurface;
 
-thread_local! {
-    static WL_COMPOSITOR: RefCell<Option<WlCompositor>> = RefCell::new(None);
+#[derive(Default)]
+pub(super) struct WlCompositorManager {
+    compositor: Option<WlCompositor>
 }
 
-pub struct WlCompositorManager {}
+impl WlCompositorManager {
+    pub fn create_surface(&self) -> Result<Surface, ()> {
+        use wayland_client::Interface; // for ::NAME
+        let compositor = self.compositor.as_ref().expect(&format!(
+            "No WlCompositor registered. Make sure your compositor supports the {} protocol",
+            WlCompositor::NAME
+        ));
+        compositor.create_surface(NewProxy::implement_dummy)
+    }
+}
 
 impl GlobalImplementor<WlCompositor> for WlCompositorManager {
     fn new_global(&mut self, new_proxy: NewProxy<WlCompositor>) -> WlCompositor {
         let res = new_proxy.implement_dummy();
 
-        WL_COMPOSITOR.with(|wl_compositor| {
-            *wl_compositor.borrow_mut() = Some(res.clone());
-        });
+        if self.compositor.replace(res.clone()).is_some() {
+            use wayland_client::Interface; // for ::NAME
+            panic!(
+                "{} already registered. Multiple compositors are not supported",
```

Comment:
This should honestly never happen, and we shouldn't bother dealing with this. Just replace it always and don't mind what was there before.